design/baremetal-operator/bmh_non-bootable_iso.md

```diff
@@ -0,0 +1,153 @@
+# Add non-bootable iso attach API to BareMetalHost
+
+## Status
+
+Implementable
+
+## Summary
+
+Add a new Custom Resource(CR), `DataImage`, to the cluster to attach a
+generic non-bootable ISO (a CD "ISO 9660" image) image to a provisioned
+or externallyProvisioned BMH using Ironic.
+
+## Motivation
+
+In certain scenarios it is desirable to attach a non-bootable iso image
+to a system after provisioning has finished, for example:
+
+A configuration ISO can be used to provide first-boot configuration to a host
+deployed from a standard image in advance. An operator may have a pool of
+powered off hosts with operating system already provisioned. Then, adding
+a host to the cluster boils down to powering it on with a configuration ISO provided.
+
+### Goals
+
+Expose the Ironic API to attach non-bootable iso images via Metal3.
+
+### Non-Goals
+
+* Supporting non-ISO images i.e. media types other than CD/DVD
+* Supporting drivers other than Redfish and its derived drivers.
+
+## Proposal
+
+### BMO Proposal
+
+Add a new Custom Resource(CR), `DataImage` (with same `name` and `namespace` as the BMH), which contains the spec containing details of the
+non-bootable iso image, example :
+
+#### Design spec
+
+```yaml
+apiVersion: metal3.io/v1alpha1
+kind: DataImage
+metadata:
+  name: metal-worker-0
+  namespace: metal-cluster
+spec:
+  url: http://1.2.3.4/image.iso
```

Comment:
Will this URL be used directly with the host or will this image be downloaded and served by ironic similar to how BMH handles images today?

Reply:
Are you talking about image registry ? As specified in the proposal, the image should be an ISO image ( i.e. CD "ISO 9660" image), other than that it can be served over network or be downloaded into a registry.

Reply:
There is an internal cache in Ironic for this process as well IIRC.